apps/src/javalab/JavalabView.jsx

```diff
@@ -107,6 +138,59 @@ class JavalabView extends React.Component {
     });
   };
 
+  handleWidthResize = desiredWidth => {
+    let newWidth = Math.max(100, Math.min(desiredWidth, 600));
+    this.props.setLeftWidth(newWidth);
+
+    this.updateLayoutThrottled(newWidth);
+  };
+
+  updateLayout = width => {
+    const visualizationColumnHeight = $(window).height() - 135;
+    const visualizationTop = this.props.topInstructionsHeight;
+    const sliderHeight = 60;
+    let constrainVisualizationWidth =
+      visualizationColumnHeight - visualizationTop;
+    if (this.props.appType === 'neighborhood') {
+      constrainVisualizationWidth -= sliderHeight;
+    }
+    let newVizWidth = Math.min(constrainVisualizationWidth, width);
+
+    let scale = newVizWidth / 800;
+    if (scale < 0) {
+      // Avoid inverting.
+      scale = 0;
+    }
+    getStore().dispatch(setVisualizationScale(scale));
+
+    const cssScale = `scale(${scale})`;
+
+    if (this.props.appType === 'neighborhood') {
+      $('#svgMaze').css('transform', cssScale);
+    } else if (this.props.appType === 'theater') {
+      $('#theater-container').css('transform', cssScale);
+    }
+
+    const cssWidth = width + 'px';
+    const cssWidthContent = newVizWidth + 'px';
+    $('#visualization').css({
+      'max-width': cssWidth,
+      'max-height': cssWidthContent,
+      height: cssWidthContent,
+      left: (width - newVizWidth) / 2 + 'px'
+    });
+
+    $('#page-small-footer .small-footer-base').css('max-width', width - 13);
+  };
+
+  updateLayoutThrottled = _.throttle(this.updateLayout, 100);
```

Comment:
Mike and I had a long conversation about this level of throttling. We're in the process of getting a low power chromebook to try it out on, but were able to simulate a similar environment on my low power laptop. When we take the throttling way down (say around 10ms), the lagginess feels like a more expected experience than the introduced lagginess by the 100ms throttling. The overall experience was the same - the panel would jump ~every 100ms no matter what the throttling was set at. However, with 100ms throttling, we'd see the maze panel jump into the editor which felt like a very broken experience. With 10ms throttling (or similar), the maze panel would still jump to its new size, but would not have components displaying over each other.

Reply:
Our long history of throttling for resizes was primarily for resizing browser windows themselves.  Things might have changed, but at one point it seemed to give the browser more time to render its own frame if we didn't try to re-render the page content relentlessly.  I'll bump this to 30 fps (~33ms) for now, and we can see how things feel after that.